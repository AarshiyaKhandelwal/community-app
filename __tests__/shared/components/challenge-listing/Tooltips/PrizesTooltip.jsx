import React from 'react';
import _ from 'lodash';
import Renderer from 'react-test-renderer/shallow';
import TU from 'react-dom/test-utils';
import PrizesTooltip from 'components/challenge-listing/Tooltips/PrizesTooltip';

const mockDatas = [
  {
    challenge: {
      prizes: [1000, 500],
      prize: [1000, 500],
      reliabilityBonus: 300,
    },
  }, {
    challenge: {
      challengeId: '1',
      prize: [1000, 500],
      reliabilityBonus: 0,
    },
  },
];

test('Matches shallow shapshot', () => {
  const renderer = new Renderer();
  _.forEach(mockDatas, (data) => {
    renderer.render((
      <PrizesTooltip {...data}>
        <div className="mock-class">abcedfghik</div>
      </PrizesTooltip>
    ));
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

class Wrapper extends React.Component {
  componentDidMount() {}

  render() {
    return (<PrizesTooltip {...this.props}>
      <div className="mock-class">abcedfghik</div>
    </PrizesTooltip>);
  }
}

test('handle events', () => {
  let instance = TU.renderIntoDocument((<Wrapper {...mockDatas[0]} />));
  let matches = TU.findAllInRenderedTree(instance, item =>
    item && item.className && item.className.match('mock-class'));
  expect(matches).toHaveLength(1);
  TU.Simulate.mouseEnter(matches[0].parentNode);

  instance = TU.renderIntoDocument((<Wrapper {...mockDatas[1]} />));
  matches = TU.findAllInRenderedTree(instance, item =>
    item && item.className && item.className.match('mock-class'));
  expect(matches).toHaveLength(1);
  TU.Simulate.mouseEnter(matches[0].parentNode);
});
