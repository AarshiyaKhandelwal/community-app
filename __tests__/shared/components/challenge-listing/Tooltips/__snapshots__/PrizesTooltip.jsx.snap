// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<Tooltip
  align={Object {}}
  className=""
  content={
    <Tip
      challenge={Object {}}
      isLoaded={true}
    />
  }
  defaultVisible={false}
  onTooltipHover={[Function]}
  placeArrow={[Function]}
  position="top"
>
  <div
    className="mock-class"
  >
    abcedfghik
  </div>
</Tooltip>
`;

exports[`Matches shallow shapshot 2`] = `
<Tooltip
  align={Object {}}
  className=""
  content={
    <Tip
      challenge={Object {}}
      isLoaded={true}
    />
  }
  defaultVisible={false}
  onTooltipHover={[Function]}
  placeArrow={[Function]}
  position="top"
>
  <div
    className="mock-class"
  >
    abcedfghik
  </div>
</Tooltip>
`;
